lnbits/static/js/event-reactions.js

```diff
@@ -0,0 +1,516 @@
+function eventReactionWebocket(event_id, event_function) {
+  localUrl = ''
+  console.log(event_id)
+  if (location.protocol !== 'http:') {
+    localUrl =
+      'wss://' + location.host + '/api/v1/ws/' + event_id
+  } else {
+    localUrl =
+      'ws://' + location.host + '/api/v1/ws/' + event_id
+  }
+  console.log(localUrl)
+  connection = new WebSocket(localUrl)
+  connection.onmessage = function (e) {
+    event_function()
+  }
+}
+function eventReactionBothSidesConfetti() {
+  document.getElementById('vue').disabled = true
+  var end = Date.now() + 2 * 1000
+  var colors = ['#FFD700', '#ffffff']
+  function frame() {
+    confetti({
+      particleCount: 2,
+      angle: 60,
+      spread: 55,
+      origin: {x: 0},
+      colors: colors,
+      zIndex: 999999
+    })
+    confetti({
+      particleCount: 2,
+      angle: 120,
+      spread: 55,
+      origin: {x: 1},
+      colors: colors,
+      zIndex: 999999
+    })
+    if (Date.now() < end) {
+      requestAnimationFrame(frame)
+    } else {
+      document.getElementById('vue').disabled = false
+    }
+  }
+  frame()
+}
+function eventReactionFireworksConfetti() {
+  var duration = 15 * 1000
+  var animationEnd = Date.now() + duration
+  var defaults = {startVelocity: 30, spread: 360, ticks: 60, zIndex: 0}
+
+  function randomInRange(min, max) {
+    return Math.random() * (max - min) + min
+  }
+
+  var interval = setInterval(function () {
+    var timeLeft = animationEnd - Date.now()
+
+    if (timeLeft <= 0) {
+      return clearInterval(interval)
+    }
+
+    var particleCount = 5 * (timeLeft / duration)
+    // since particles fall down, start a bit higher than random
+    confetti({
+      ...defaults,
+      particleCount,
+      origin: {x: randomInRange(0.1, 0.3), y: Math.random() - 0.2}
+    })
+    confetti({
+      ...defaults,
+      particleCount,
+      origin: {x: randomInRange(0.7, 0.9), y: Math.random() - 0.2}
+    })
+  }, 250)
+}
+
+!(function (t, e) {
```

Comment:
is this a library or what is this? did you write this?

Reply:
Well used little  confetti library, thats easy to extend, such as in the above couple of functions.